pkg/handler/v2/alertmanager/alertmanager_test.go

```diff
@@ -0,0 +1,410 @@
+	kubermaticv1 "k8c.io/kubermatic/v2/pkg/crd/kubermatic/v1"
+	"k8c.io/kubermatic/v2/pkg/handler/test"
+	"k8c.io/kubermatic/v2/pkg/handler/test/hack"
+
+	corev1 "k8s.io/api/core/v1"
+	ctrlruntimeclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	testAlertmanagerConfigSecretName = "test-alertmanager"
+	testAlertmanagerConfig           = `
+alertmanager_config: |
+  global:
+    smtp_smarthost: 'localhost:25'
+    smtp_from: 'test@example.org'
+  route:
+    receiver: "test"
+  receivers:
+    - name: "test"
+      email_configs:
+      - to: 'test@example.org'
+`
+)
+
+func TestGetEndpoint(t *testing.T) {
+	t.Parallel()
+	testCases := []struct {
+		Name                      string
+		ProjectID                 string
+		ClusterID                 string
+		ExistingKubermaticObjects []ctrlruntimeclient.Object
+		ExistingAlertmanager      *kubermaticv1.Alertmanager
+		ExistingConfigSecret      *corev1.Secret
+		ExistingAPIUser           *apiv1.User
+		ExpectedResponse          apiv2.Alertmanager
+		ExpectedHTTPStatus        int
+	}{
+		{
+			Name:      "scenario 1: get alertmanager that belongs to the given cluster",
+			ProjectID: test.GenDefaultProject().Name,
+			ClusterID: test.GenDefaultCluster().Name,
+			ExistingKubermaticObjects: test.GenDefaultKubermaticObjects(
+				test.GenTestSeed(),
+				test.GenDefaultCluster(),
+			),
+			ExistingAlertmanager: test.GenAlertmanager(test.GenDefaultCluster().Status.NamespaceName,
+				testAlertmanagerConfigSecretName),
+			ExistingConfigSecret: test.GenAlertmanagerConfigSecret(testAlertmanagerConfigSecretName,
+				test.GenDefaultCluster().Status.NamespaceName,
+				[]byte(testAlertmanagerConfig)),
+			ExistingAPIUser:    test.GenDefaultAPIUser(),
+			ExpectedHTTPStatus: http.StatusOK,
+			ExpectedResponse: apiv2.Alertmanager{
+				Spec: apiv2.AlertmanagerSpec{
+					Config: []byte(testAlertmanagerConfig),
+				},
+			},
+		},
+		{
+			Name:      "scenario 2: try to get alertmanager but alertmanager is not found",
+			ProjectID: test.GenDefaultProject().Name,
+			ClusterID: test.GenDefaultCluster().Name,
+			ExistingKubermaticObjects: test.GenDefaultKubermaticObjects(
+				test.GenTestSeed(),
+				test.GenDefaultCluster(),
+			),
+			ExistingAPIUser:    test.GenDefaultAPIUser(),
+			ExpectedHTTPStatus: http.StatusNotFound,
+		},
+		{
+			Name:      "scenario 3: try to get alertmanager but alertmanager config secret is not found",
+			ProjectID: test.GenDefaultProject().Name,
+			ClusterID: test.GenDefaultCluster().Name,
+			ExistingKubermaticObjects: test.GenDefaultKubermaticObjects(
+				test.GenTestSeed(),
+				test.GenDefaultCluster(),
+			),
+			ExistingAlertmanager: test.GenAlertmanager(test.GenDefaultCluster().Status.NamespaceName,
+				testAlertmanagerConfigSecretName),
+			ExistingAPIUser:    test.GenDefaultAPIUser(),
+			ExpectedHTTPStatus: http.StatusNotFound,
+		},
+		{
+			Name:      "scenario 4: user john can not get alertmanager that belongs to bob's cluster",
```

Comment:
you can add a test that Admin can get an alertManager that  belongs to bobs cluster

Reply:
Currently, this endpoint doesn't work for admins because I didn't embed a `privilegedAlertmanagerProvider` there, and it only works for cluster owners/editors/viewers.

Reply:
ahh ok. We should plan it as well for the future, because "admins" in KKP are like gods and we make little backdoors for them everywhere. Its not pretty and hopefully when we make our big RBAC/user management rehaul it will get removed

Reply:
Yeah, this definitely makes sense, I have created an issue for this: #6966.